apps/site/lib/site_web/controllers/alert_controller.ex

```diff
@@ -30,6 +30,22 @@ defmodule SiteWeb.AlertController do
     check_cms_or_404(conn)
   end
 
+  # TODO revisit this later to see if it is used/needed
+  @spec show_by_stop(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def show_by_stop(conn, %{"stop_id" => stop_id}) do
+    alerts = Repo.by_stop_id(stop_id)
+    json(conn, alerts)
+  end
+
+  @spec show_by_routes(Plug.Conn.t(), map) :: Plug.Conn.t()
+  def show_by_routes(conn, _) do
```

Comment:
You can pattern match on `query_params` if you wanna -- `%Plug.Conn{query_params: query_params} = conn`

Anyway, please add a function clause for handling when the `route_ids` query param is missing.